src/pages/Search/AdvancedSearchFilters.tsx

```diff
@@ -451,6 +453,20 @@ function AdvancedSearchFilters() {
         .map(getTagNamesFromTagsLists)
         .flat();
 
+    const [searchAdvancedFiltersForm] = useOnyx(ONYXKEYS.FORMS.SEARCH_ADVANCED_FILTERS_FORM);
+    const [currentUserLogin] = useOnyx(ONYXKEYS.SESSION, {selector: (session) => session?.email});
+    const [debouncedSearchTerm] = useDebouncedState('');
+
+    const {sections: workspaces} = useWorkspaceList({
+        policies,
+        currentUserLogin,
+        shouldShowPendingDeletePolicy: false,
+        selectedPolicyID: searchAdvancedFiltersForm?.policyID,
+        searchTerm: debouncedSearchTerm,
+    });
+
+    const isWorkspaceNameStillPresent = (filterTitle: string | undefined) => workspaces.some((section) => section.data?.some((item) => item.text === filterTitle));
```

Comment:
I added this to prevent scenarios like this:

https://github.com/user-attachments/assets/fffa1b2b-565d-4d7e-aaad-d567244bfc59

Reply:
I tried not to disrupt the logic we've been using so far to avoid introducing new bugs.
If you think it's better to change it, I can do that, but my solution seems quite clear to me and doesn't disrupt the previous logic, the only issue with which was showing the filter when it wasn't necessarily needed.

Reply:
I think it makes sense. I know it's behind the beta here, but I was afraid of breaking something that was already working in it. But now I understand what you mean, and it seems like a reasonable fix to me.